frontend/pulse3d/src/pages/users-info.js

```diff
@@ -0,0 +1,254 @@
+import DashboardLayout from "@/components/layouts/DashboardLayout";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import Paper from "@mui/material/Paper";
+import { useState, useEffect } from "react";
+import Row from "@/components/admin/UsersRow";
+import styled from "styled-components";
+import ModalWidget from "@/components/basicWidgets/ModalWidget";
+
+const PageContainer = styled.div`
+  width: 80%;
+`;
+
+const Pagination = styled.div`
+  width: 100%;
+  background-color: var(--dark-gray);
+  display: flex;
+  justify-content: right;
+  padding: 1rem 1rem;
+`;
+
+const Container = styled.div`
+  display: flex;
+  position: relative;
+  justify-content: start;
+  padding: 0% 3% 3% 3%;
+  flex-direction: column;
+  margin-top: 1rem;
+`;
+
+export default function UserInfo() {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [rows, setRows] = useState([]);
+  const [currentRows, setCurrentRows] = useState([]);
+  const [users, setUsers] = useState([]);
+  const [actionAlertVisible, setActionAlertVisible] = useState(false);
+  const [modalMessage, setModalMessage] = useState();
+  const [actionToPreform, setActionToPerform] = useState("");
+  const [confirm, setConfirm] = useState(false);
+  const [emailToEdit, setEmailToEdit] = useState("");
+
+  const deleteUser = async () => {
```

Comment:
This and `deactivateUser` are almost identical, I think these could be one method that takes the action as a paremeter